### PR TITLE
Soen68 back end assign patient doctors

### PR DESCRIPTION
**Pull request by jfv-2000**
# Description

Implements/fixes [SOEN-68](https://soen390team8.atlassian.net/jira/software/projects/SOEN/boards/1?selectedIssue=SOEN-23)

- Back-end implementation which allows an admin to assign a patient with a doctor
- Test implementation for the route that assigns a patient with a doctor
- Test done with Supertest to text express routes
- Patient model added for sequelize
- If a user registers as a patient, automatically add that user to the Patient table
- Code formatted with Prettier (Default)

- Bug related to CI build due to open handles during testing
- Bug fixed by restructuring the index.js

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit testing for the route

# Screenshots

|         Original          |         Updated          |
| :-----------------------: | :----------------------: |
| ** original screenshot ** | ** updated screenshot ** |

### CODE REVIEW

**Review by NicholasYip — APPROVED** [review]
Juicy work, love the tests

**Review by stevenvuong00 — APPROVED** [review]
Looks nice, good job on the tests. Great work!

File: backend/models/patient.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        autoIncrement: true,
+        type: Sequelize.INTEGER
+    },
+    Height: {
+        type: Sequelize.STRING
+    },
+    Weight: {
+        type: Sequelize.STRING
+    },
+    isPrioritized: {
```
Comment (amqkv) [inline backend/models/patient.js]: For consistency, this should be PascalCase to match the database

File: backend/controllers/admin-controller.js [hunk 2]
```diff
@@ -16,15 +22,46 @@ function updateRole(req, res) {
       }
     )
       .then((result) => {
-        console.log('Role successfully updated !');
-        res.status(200).send('Role successfully updated !');
+        console.log("Role successfully updated !");
```
Comment (amqkv) [inline backend/controllers/admin-controller.js]: Maybe remove unneeded console.logs

File: backend/controllers/admin-controller.js [hunk 3]
```diff
@@ -16,15 +22,46 @@ function updateRole(req, res) {
...
+    },
+    {
+      where: {
+        PatientId: req.body.patientId,
+      },
+    }
+  )
+    .then((patient) => {
+      if (patient[0]) {
+        console.log("Patient has been assigned to doctor");
```
Comment (amqkv) [inline backend/controllers/admin-controller.js]: This too

File: backend/controllers/auth-controller.js [hunk 4]
```diff
@@ -19,9 +20,21 @@ function register(req,res) {
         Role: req.body.accountRole,
         Confirmed: req.body.accountRole === constants.ROLE.PATIENT
     }).then(user => {
-        if(user) {
-            console.log("new user created");
-            res.json(user)
+        if (user) {
+            console.log("New user created");
```
Comment (amqkv) [inline backend/controllers/auth-controller.js]: And this

File: backend/controllers/auth-controller.js [hunk 5]
```diff
@@ -19,9 +20,21 @@ function register(req,res) {
...
+            res.json(user);
+            if (user.Role === constants.ROLE.PATIENT) {
+                const newPatient = Patient.create({
+                    User_AccountId: user.AccountId
+                }).then(patient => {
+                    if (patient) {
+                        console.log("Patient added to table");
```
Comment (amqkv) [inline backend/controllers/auth-controller.js]: Esto también

**Review by amqkv — APPROVED** [review]
Scrumptious work 👁️👄👁️👍

**Pull request merged**